core/templates/services/state-interaction-stats.service.ts

```diff
@@ -111,24 +106,26 @@ export class StateInteractionStatsService {
           exploration_id: explorationId,
           state_name: state.name,
         }))
-      .toPromise().then(response => <IStateRulesStats>{
-        exploration_id: explorationId,
-        state_name: state.name,
-        visualizations_info: response.visualizations_info.map(info => ({
-          addressed_info_is_supported: info.addressed_info_is_supported,
-          data: info.data.map(datum => <IAnswerData>{
-            answer: this.getReadableAnswerString(state, datum.answer),
-            frequency: datum.frequency,
-            is_addressed: (
-              info.addressed_info_is_supported ?
+      .toPromise().then(response => {
+        this.cachedStats = {
+          exploration_id: explorationId,
+          state_name: state.name,
+          visualizations_info: response.visualizations_info.map(vizInfo => ({
+            id: vizInfo.id,
+            options: vizInfo.options,
+            addressed_info_is_supported: vizInfo.addressed_info_is_supported,
+            data: vizInfo.data.map(datum => <IAnswerData>{
+              answer: this.getReadableAnswerString(state, datum.answer),
+              frequency: datum.frequency,
+              is_addressed: vizInfo.addressed_info_is_supported ?
                 this.answerClassificationService
                   .isClassifiedExplicitlyOrGoesToNewState(
                     state.name, state, datum.answer, interactionRulesService) :
-                undefined),
-          }),
-          id: info.id,
```

Comment:
Should probably justify this deletion.

Reply:
It's not deleted, just moved to the top of the definition (to keep the simple copy-only fields gathered).

Reply:
Oh, sorry! You're right, I missed that. Apologies.